gsd/__main__.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+
+"""The GSD command line interface.
+
+To simplify ad hoc usage of :py:mod:`gsd`, this module provides a command line
+interface for interacting with GSD files. The primary entry point is a single
+command for starting a Python interpreter with a GSD file pre-loaded::
+
+    $ gsd read trajectory.gsd
```

Comment:
Add documention for the CLI in Sphinx and list the accepted arguments and resulting behavior.

Also add an example using the CLI in hoomd-examples.rst. I think you can use the ipython shell magic in a ..ipython block so the example will execute when the docs build.

Reply:
I did this with an IPython block, but there's a small issue: since this opens an actual interpreter, it freezes the doc build until you send a SIGINT, so I just changed it to print the help. The second-to-last commit uses a console block instead of an ipython block if you want to skip the execution entirely, but I don't see a great way around the interpreter issue.

Reply:
OK, let's just go without the ipython block then. It isn't critical for this example output to update with each new doc build and you can provide a better example with a static block.